### ISSUE TITLE
Encoding and Spanish Characters

### ISSUE THREAD

**Issue by ramnathv** [viking/r-yaml#10]
I am having trouble getting `yaml` to work under `Windows-1252` encoding. Here is some code to reproduce the error. Any thoughts on how to resolve this?

``` S
library(yaml)
yaml.load('title       : Introducción a R
subtitle    : Clases de objetos en R
author      : Arturo Chian, Raúl Panduro
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}'
)
```

**Comment by yihui** [comment]
I think this is the same issue as #6.

**Comment by ofurkusi** [comment]
The issues are related, but this is not quite the same issue as https://github.com/viking/r-yaml/issues/6.

While ``yaml.load`` does not seem to be bothered whether the input string is *marked* as UTF-8 or not (i.e. not complaining if it is set to *unknown* as in the case of ``yaml.load_file``), it does give an error if the input string is *not truly* UTF-8 encoded. For Windows users this means that passing a string containing special characters manually to ``yaml.load`` will not work as they are latin1 encoded, e.g.:
````
> yaml.load("þæö")
Error in yaml.load("þæö") : 
  Reader error: invalid leading UTF-8 octet: #FE at 0
````

Because ``yaml.load`` is picky about the encoding it receives - and it should be - it must in my opinion check the encoding of the input string before attempting to parse it and at least give the user an error that helps with debugging.

A more helpful solution is to automatically attempt to convert the input to UTF-8. Something along these lines.
````
yaml.load <- function (string, as.named.list = TRUE, handlers = NULL) 
{
  string <- enc2utf8(string)
  .Call("yaml.load", string, as.named.list, handlers, PACKAGE = "yaml")
}
````
This works whether the string is UTF-8 or not.

Issue https://github.com/viking/r-yaml/issues/6 however remains.
````
> yaml.load("þæö")
[1] "Ã¾Ã¦Ã¶"
````

**Comment by yihui** [comment]
FYI I submitted a fix in #32.
